File: src/AtCoder/abc286/a.py
N, P, Q, R, S = map(int, input().split())
A = list(map(int, input().split()))

P -= 1
Q -= 1
R -= 1
S -= 1

for d in range(0, Q-P+1):
    A[P+d], A[R+d] = A[R+d], A[P+d]

print(*A)

File: src/AtCoder/abc286/a.rs
use itertools::Itertools;
use proconio::input;
use proconio::marker::*;

fn main() {
    input! {
        n: usize,
        mut p: usize,
        mut q: usize,
        mut r: usize,
        mut s: usize,
        mut a: [usize; n],
    }
    p -= 1;
    q -= 1;
    r -= 1;
    s -= 1;

    for d in 0..=(q-p) {
        a.swap(p+d, r+d);
    }
    println!("{}", a.iter().join(" "));
}
